pi/mlir/utils.py

```diff
@@ -192,8 +192,8 @@ def to_value_tensor_type_bound(self):
         type_attr = TypeAttr.get(self.to_value_tensor_type())
         return DictAttr.get({"torch.type_bound": type_attr})
 
-    def to(self, dtype_: dtype):
-        self.dtype = dtype_
+    def to(self, dtype: dtype):
```

Comment:
nit: the arg itself should be named `dtype_` so it doesn't shadow the type `dtype`. in small functions it doesn't matter of course but just in general it helps so you don't lose the type.

Reply:
I figured that this was why you didn't name them the same thing - this is just one solution, another might be to rename the python-binded class name for the DType enum though that requires a bit of tracking. Any suggestions/is this something you want me to address?

Reply:
Yes this is being called like `a.to(dtype=torch.float32)` in some test cases.